plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -0,0 +1,1063 @@
+/*
+ * Copyright (C) 2023 Crag Wang <crag.wang@dell.com>
+ * Copyright (C) 2023 Mediatek Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <glib/gstdio.h>
+#include <linux/i2c-dev.h>
+
+#include "fu-mediatek-scaler-device.h"
+
+#define DDC_DATA_LEN_DFT	0x80
+#define DDC_OPCODE_MTK_REQ	0xCC   /* vendor specific opcode */
+#define DDC_DATA_FRAGEMENT_SIZE 0x0B   /* 11 bytes for each DDC write */
+#define DDC_DATA_PAGE_SIZE	0x1000 /* 4K bytes for each block page */
+#define DDC_RW_MAX_RETRY_CNT	10
+
+/* timeout duration in ms to i2c-dev operation */
+#define FU_MEDIATEK_SCALER_DEVICE_IOCTL_TIMEOUT 5000
+
+/* delay time before a ddc read or write */
+#define FU_MEDIATEK_SCALER_DDC_MSG_DELAY_MS 50
+
+/* interval in ms between the poll to check device status */
+#define FU_MEDIATEK_SCALER_DEVICE_POLL_INTERVAL 1000
+
+typedef enum {
+	FIRMWARE_SIZE_MAX = 0x100000,
+	FIRMWARE_OFFSET_VERSION = 0x7118,
+	FIRMWARE_OFFSET_TIMESTAMP_DATE = 0x7200,
+	FIRMWARE_OFFSET_TIMESTAMP_TIME = 0x720c,
+	FIRMWARE_TIMESTAMP_DATE_SIZE = 11,
+	FIRMWARE_TIMESTAMP_TIME_SIZE = 8,
+} FuMediatekScalerFirmware;
```

Comment:
what is this enum?  I think they could all be #defines tbh

Reply:
not really an enum for them, just categorized into a section. Okay, will tear them down.